compose/queue/api_data/taskqueue_v1beta2.py

```diff
@@ -0,0 +1,626 @@
+            raise Exception("Not all values specified.")
+
+        q = TaskQueue(**json)
+        q.stats = Stats.from_json(json["stats"])
+        q.acl = Acl.from_json(json["acl"])
+        return q
+
+
+# Taskqueue Blueprint
+taskqueue_v1beta2_api = Blueprint("taskqueue.v1beta2", __name__)
+
+
+@taskqueue_v1beta2_api.route(
+    "/<project>/taskqueues/<taskqueue>/tasks",
+    methods=["GET", "POST"])
+def tasks_api(project=None, taskqueue=None):
+    """List items or Insert into a given queue.
+
+    Args:
+        project: A string of the project to work on.
+        taskqueue: A string of the affected taskqueue.
+
+    GET:
+        Lists all non-deleted Tasks in a TaskQueue, whether or not
+        they are currently leased, up to a maximum of 100.
+
+        Returns:
+            A json object containing a kind and items attribute.
+            Where items is a list of tasks.
+
+    POST:
+        Inserts a task into an existing queue.
+
+        Body:
+            Must be a JSON Task object, where only the queueName
+            and payloadBase64 need to be specified.
+        Returns:
+            A json object of the created Task.
+    """
+    if project is None:
+        return "You must specify a project.", 400
+    elif taskqueue is None:
+        return "You must specify a taskqueue.", 400
+
+    project = project.replace("b~", "")
+    queue_key = ".".join([project, taskqueue])
+
+    # Lists all non-deleted Tasks in a TaskQueue, whether or not
+    # they are currently leased, up to a maximum of 100.
+    if request.method == "GET":
+        start = 0
+        tasks = []
+        numTasks = 100
+        while len(tasks) < numTasks:
+            tasks_needed = numTasks - len(tasks)
+            end = start + tasks_needed
+            keys = r.zrange(name=queue_key, start=start, end=end)
+            if len(keys) == 0:
+                break
+
+            for key in keys:  # Could become parallelizable
+                task_string = r.get(name=key)
+                if task_string is None:
+                    continue
+
+                task_string = task_string.decode()
+                task_json = json.loads(task_string)
+                tasks.append(task_json)
+
+            start = end + 1
+
+        return json.dumps({
+            "kind": "taskqueues#tasks",
+            "items": tasks
+        })
+
+    # Insert a task into an existing queue.
+    elif request.method == "POST":
+        task_json = json.loads(request.get_data().decode())
+
+        task = Task(**task_json)
+        task_string = json.dumps(task.to_json())
+        task_id = task.id
+
+        if r.exists(name=task_id):
+            return "Task name is invalid.", 400
+
+        p = r.pipeline()
+        p.zadd(queue_key, task.enqueueTimestamp, task_id)
+        p.set(name=task_id, value=task_string)
+        p.execute()
+
+        return task_string
+
+
+@taskqueue_v1beta2_api.route(
+    "/<project>/taskqueues/<taskqueue>/tasks/lease",
```

Comment:
Its a REST API url, this doesn't get deployed and is documented by Google there is no hiding it.